OrbitQt/OverlayWidget.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_QT_OVERLAY_WIDGET_H_
+#define ORBIT_QT_OVERLAY_WIDGET_H_
+
+#include <QEvent>
+#include <QObject>
+#include <QPainter>
+#include <QWidget>
+#include <memory>
+
+#include "OrbitBase/Logging.h"
+#include "ui_OverlayWidget.h"
+
+namespace OrbitQt {
+
+class OverlayWidget : public QWidget {
+  Q_OBJECT
+  Q_PROPERTY(bool spinning READ IsSpinning WRITE SetSpinning)
+  Q_PROPERTY(bool cancelable READ IsCancelable WRITE SetCancelable);
+  Q_PROPERTY(QString statusMessage READ GetStatusMessage WRITE SetStatusMessage)
+  Q_PROPERTY(QString buttonMessage READ GetButtonMessage WRITE SetButtonMessage);
+
+ public:
+  explicit OverlayWidget(QWidget* parent);
+
+ protected:
+  void paintEvent(QPaintEvent* /*event*/) override;
+  bool eventFilter(QObject* obj, QEvent* event) override;
+
+ private slots:
+  void on_cancelButton_clicked() { emit Cancelled(); }
+  void SetSpinning(bool value) { ui_->progressBar->setVisible(value); }
+  void SetCancelable(bool value) { ui_->cancelButton->setVisible(value); }
+  void SetStatusMessage(QString message) { ui_->messageLabel->setText(message); }
+  void SetButtonMessage(QString message) { ui_->cancelButton->setText(message); }
+
+ signals:
+  void Cancelled();
+
+ private:
+  QWidget* parent_;
```

Comment:
parent_ is like parent() but a QWidget*? What about parentWidget()?

Reply:
Good idea. I used parentWidget()